internal/store/device.go

```diff
@@ -629,6 +638,35 @@ func (s *DeviceStore) SetServiceConditions(ctx context.Context, orgId uuid.UUID,
 	})
 }
 
+func (s *DeviceStore) SetDecommission(ctx context.Context, orgId uuid.UUID, name string, request *api.DeviceDecommission) error {
+	existingRecord := model.Device{Resource: model.Resource{OrgID: orgId, Name: name}}
+	result := s.db.First(&existingRecord)
+	if result.Error != nil {
+		return ErrorFromGormError(result.Error)
+	}
+
+	if existingRecord.Spec == nil {
+		existingRecord.Spec = model.MakeJSONField(api.DeviceSpec{})
+	}
+	if existingRecord.Spec.Data.DecommissionRequested != nil {
+		return fmt.Errorf("decommission request on already decommissioning device")
+	}
+	existingRecord.Spec.Data.DecommissionRequested = request
+
+	result = s.db.Model(existingRecord).Where("resource_version = ?", lo.FromPtr(existingRecord.ResourceVersion)).Updates(map[string]interface{}{
```

Comment:
This is missing retries on resourceVersion conflict.  In a comment above I suggested using Update(), which already has this.